File: dmn-core/src/test/resources/tck/cl2/expected/dmn/0007-simpletable-p2/ApprovalStatusRuleOutput.java

import java.util.*;

@javax.annotation.Generated(value = {"decisionTableRuleOutput.ftl", "ApprovalStatus"})
public class ApprovalStatusRuleOutput extends com.gs.dmn.runtime.RuleOutput {
    private String approvalStatus;
    private Integer approvalStatusPriority;

    public ApprovalStatusRuleOutput(boolean matched) {
        super(matched);
    }

    public String getApprovalStatus() {
        return this.approvalStatus;
    }
    public void setApprovalStatus(String approvalStatus) {
        this.approvalStatus = approvalStatus;
    }

    public Integer getApprovalStatusPriority() {
        return this.approvalStatusPriority;
    }
    public void setApprovalStatusPriority(Integer approvalStatusPriority) {
        this.approvalStatusPriority = approvalStatusPriority;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        ApprovalStatusRuleOutput other = (ApprovalStatusRuleOutput) o;
        if (this.getApprovalStatus() != null ? !this.getApprovalStatus().equals(other.getApprovalStatus()) : other.getApprovalStatus() != null) return false;
        if (this.getApprovalStatusPriority() != null ? !this.getApprovalStatusPriority().equals(other.getApprovalStatusPriority()) : other.getApprovalStatusPriority() != null) return false;

        return true;
    }

    @Override
    public int hashCode() {
        int result = 0;
        result = 31 * result + (this.getApprovalStatus() != null ? this.getApprovalStatus().hashCode() : 0);
        result = 31 * result + (this.getApprovalStatusPriority() != null ? this.getApprovalStatusPriority().hashCode() : 0);

        return result;
    }

    public String toString() {
        StringBuilder result_ = new StringBuilder("(matched=" + isMatched());
        result_.append(String.format(", approvalStatus='%s'", approvalStatus));
        result_.append(")");
        return result_.toString();
    }

    @Override
    public List<com.gs.dmn.runtime.RuleOutput> sort(List<com.gs.dmn.runtime.RuleOutput> matchedResults_) {
        List<com.gs.dmn.runtime.Pair<String, Integer>> approvalStatusPairs = new ArrayList<>();
        matchedResults_.forEach(matchedResult_ -> {
            approvalStatusPairs.add(new com.gs.dmn.runtime.Pair(((ApprovalStatusRuleOutput)matchedResult_).getApprovalStatus(), ((ApprovalStatusRuleOutput)matchedResult_).getApprovalStatusPriority()));
        });
        approvalStatusPairs.sort(new com.gs.dmn.runtime.PairComparator());

        List<com.gs.dmn.runtime.RuleOutput> result_ = new ArrayList<com.gs.dmn.runtime.RuleOutput>();
        for(int i=0; i<matchedResults_.size(); i++) {
            ApprovalStatusRuleOutput output_ = new ApprovalStatusRuleOutput(true);
            output_.setApprovalStatus(approvalStatusPairs.get(i).getLeft());
            output_.setApprovalStatusPriority(approvalStatusPairs.get(i).getRight());
            result_.add(output_);
        }
        return result_;
    }
}

File: dmn-test-cases/tck/1.1/cl2/0007-simpletable-P2/translator/expected/kotlin/dmn/ApprovalStatusRuleOutput.kt

@javax.annotation.Generated(value = ["decisionTableRuleOutput.ftl", "ApprovalStatus"])
class ApprovalStatusRuleOutput(matched: Boolean) : com.gs.dmn.runtime.RuleOutput(matched) {
    @com.fasterxml.jackson.annotation.JsonProperty("ApprovalStatus")
    var approvalStatus: String? = null
    var approvalStatusPriority: Int? = 0

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (javaClass != o?.javaClass) return false

        val other = o as ApprovalStatusRuleOutput
        if (if (this.approvalStatus != null) this.approvalStatus != other.approvalStatus else other.approvalStatus != null) return false
        if (if (this.approvalStatusPriority != null) this.approvalStatusPriority != other.approvalStatusPriority else other.approvalStatusPriority != null) return false

        return true
    }

    override fun hashCode(): Int {
        var result = 0
        result = 31 * result + (if (this.approvalStatus != null) this.approvalStatus.hashCode() else 0)
        result = 31 * result + (if (this.approvalStatusPriority != null) this.approvalStatusPriority.hashCode() else 0)

        return result
    }

    override fun toString(): String {
        val result_ = StringBuilder("(matched=" + isMatched())
        result_.append(String.format(", approvalStatus='%s'", approvalStatus))
        result_.append(")")
        return result_.toString()
    }

    override fun sort(matchedResults_: MutableList<com.gs.dmn.runtime.RuleOutput>): MutableList<com.gs.dmn.runtime.RuleOutput> {
        val approvalStatusPairs: MutableList<com.gs.dmn.runtime.Pair<String?, Int?>> = ArrayList()
        matchedResults_.forEach({ (it as ApprovalStatusRuleOutput)
            approvalStatusPairs.add(com.gs.dmn.runtime.Pair(it.approvalStatus, it.approvalStatusPriority))
        })
        approvalStatusPairs.sortWith(com.gs.dmn.runtime.PairComparator())

        val result_: MutableList<com.gs.dmn.runtime.RuleOutput> = ArrayList<com.gs.dmn.runtime.RuleOutput>()
        for(i in 0 until matchedResults_.size) {
            var output_ = ApprovalStatusRuleOutput(true)
            output_.approvalStatus = approvalStatusPairs.get(i).getLeft()
            output_.approvalStatusPriority = approvalStatusPairs.get(i).getRight()
            result_.add(output_)
        }
        return result_
    }
}
